apps/wallet/src/ui/app/redux/slices/transaction-requests/index.ts

```diff
@@ -65,10 +66,105 @@ export const loadTransactionResponseMetadata = createAsyncThunk<
     }
 );
 
+export type TxnMetaResponse = {
+    coinSymbol?: string | null;
+    amount?: number | null;
+    objectId?: string;
+    isListing?: boolean;
+} | null;
+
+const getRequestCost = (
+    txEffects: TransactionEffects,
+    address: string
+): TxnMetaResponse | null => {
+    const events = txEffects?.events || [];
+
+    let sumCoin = 0;
+    const coinMeta: {
+        objectId?: string;
+        amount?: number;
+        coinSymbol?: string;
+        isListing?: boolean;
+    } = {};
+
+    for (const event of events) {
+        if (
+            'coinBalanceChange' in event ||
+            'transferObject' in event ||
+            'moveEvent' in event
+        ) {
+            // Get the amount in move events
+            if (
+                event.moveEvent?.sender === address &&
+                event.moveEvent?.fields?.price
+            ) {
+                sumCoin += event.moveEvent.fields.price;
+                coinMeta.amount = sumCoin;
+                coinMeta.coinSymbol = '0x2::sui::SUI';
+                coinMeta.isListing = true;
+            }
+
+            // aggregate coin balance for pay
+            if (
+                event.coinBalanceChange?.changeType === 'Pay' &&
+                event.coinBalanceChange?.owner.AddressOwner === address
+            ) {
+                sumCoin += event.coinBalanceChange?.amount || 0;
+                coinMeta.amount = sumCoin;
+                coinMeta.coinSymbol = event.coinBalanceChange?.coinType
```

Comment:
Same as above - coinType not coinSymbol

Also why not just coinMeta.coinSymbol = event.coinBalanceChange?.coinType || ''